src/main/java/org/cardanofoundation/job/schedules/AddressTxCountSchedule.java

```diff
@@ -53,21 +55,22 @@ public void syncAddressTxCount() {
     if (latestBlock.isEmpty()) {
       return;
     }
-    final long currentMaxBlockNo =
-        Math.min(
-            addressTxAmountRepository.getMaxBlockNoFromCursor(), latestBlock.get().getBlockNo());
+
+    final long currentMaxSlotNo =
+        Math.min(latestBlock.get().getSlotNo(), addressTxAmountRepository.getMaxSlotNoFromCursor());
+
     final Integer checkpoint = redisTemplate.opsForValue().get(addressTxCountCheckPoint);
-    if (Objects.isNull(checkpoint)) {
+    if (Objects.isNull(checkpoint) || addressTxCountRepository.count() == 0) {
       init();
-    } else if (currentMaxBlockNo > checkpoint.longValue()) {
-      update(checkpoint.longValue(), currentMaxBlockNo);
+    } else if (currentMaxSlotNo > checkpoint.longValue()) {
+      update(checkpoint.longValue(), currentMaxSlotNo);
     }
 
-    // Update the checkpoint to the currentMaxBlockNo - 2160 to avoid missing any data when node
+    // Update the checkpoint to the currentMaxSlotNo - 43200 to avoid missing any data when node
     // rollback
     redisTemplate
         .opsForValue()
-        .set(addressTxCountCheckPoint, Math.max((int) currentMaxBlockNo - 2160, 0));
+        .set(addressTxCountCheckPoint, Math.max((int) currentMaxSlotNo - 43200, 0));
```

Comment:
Why - 43200? Could you replace it with a constant to avoid magic numbers.

Reply:
I have added a comment on this. Generally, in the case of a rollback, we need to shift the checkpoint to ensure the integrity of the data. Note that 43,200 slots equal 12 hours.

Reply:
Yes, but still if we want to change it any time, then we need to change multiple places. A constant would be help there a lot